tools/cellxgene_census_builder/src/cellxgene_census_builder/build_soma/manifest.py

```diff
@@ -96,10 +98,43 @@ def load_manifest_from_CxG() -> List[Dataset]:
     return response
 
 
-def load_manifest(manifest_fp: Optional[Union[str, io.TextIOBase]] = None) -> List[Dataset]:
+def load_blocklist(dataset_id_blocklist_uri: str | None) -> Set[str]:
+    blocked_dataset_ids: Set[str] = set()
+    if dataset_id_blocklist_uri is None:
+        logging.info("No dataset blocklist specified - skipping")
+        return blocked_dataset_ids
+
+    with fsspec.open(dataset_id_blocklist_uri, "rt") as fp:
+        for line in fp:
+            line = line.strip()
+            if len(line) == 0 or line.startswith("#"):
+                # strip blank lines and comments (hash is never in a UUID)
+                continue
+            blocked_dataset_ids.add(line)
+
+        logging.info(f"Dataset blocklist found, containing {len(blocked_dataset_ids)} ids.")
+
+    return blocked_dataset_ids
+
+
+def apply_blocklist(datasets: List[Dataset], dataset_id_blocklist_uri: str | None) -> List[Dataset]:
+    try:
+        blocked_dataset_ids = load_blocklist(dataset_id_blocklist_uri)
+        return list(filter(lambda d: d.dataset_id not in blocked_dataset_ids, datasets))
+
+    except FileNotFoundError:
+        # Blocklist may not exist, so just skip the filtering in this case
+        logging.info("No dataset blocklist found - skipping")
```

Comment:
I think we should error on this instead. If a blocklist is specified and it doesn't exist, that seems like an configuration error. And means datasets are included unexpectedly. The blocklist will be immediately available upon this PR being merged.

Reply:
will add